yapapi/strategy.py

```diff
@@ -190,3 +213,34 @@ async def score_offer(self, offer: rest.market.OfferProposal) -> float:
             self._logger.debug("Decreasing score for offer %s from '%s'", offer.id, offer.issuer)
             score *= self.factor
         return score
+
+
+class StrategySupportingMidAgreementPayments(MarketStrategy):
```

Comment:
Do I understand correctly that the only thing this strategy does is:
1. Hardcode two values (`debit-note-interval-sec?` and `payment-timeout-sec?`)
2. Adds max datetime expiration to the demand builder
?

If yes - why do we need the second one, again?
And if we don't need the second one, why this strategy at all?

Reply:
This is discussed in https://github.com/golemfactory/yapapi/issues/794 (I linked to it in the description of this PR).

2. Using mid-term agreements requires setting long expiration date in the Demand. Otherwise, the provider won't enable them and we are using this strategy to enable mid-term agreements.

1. It also provides default values for `debit-note-interval-sec` and `payment-timeout-sec` negotiation and allows users to set their own values.